src/sensai/vector_model.py

```diff
@@ -389,7 +406,6 @@ def convertClassProbabilitiesToPredictions(self, df: pd.DataFrame):
         """
         modelOutputVariableName = self.getModelOutputVariableNames()[0]
         y = self._convertClassProbabilitiesToPredictions(df, predictedVariableName=modelOutputVariableName)
-        # TODO check sanity: where should we be applying post-processing?
```

Comment:
![image](https://user-images.githubusercontent.com/1295099/105399158-8af73680-5c23-11eb-8932-ab82bd2d5538.png)

Reply:
You fixed this by calling `_convertClassProbabilities` instead, but I would argue that it is rather unexpected for this method to do anything but what its name suggests, which should not include the post-processing. This is also a breaking change, because the previous public method did not apply post-processing.
Therefore, I would propose to keep only the public method (and inline the private method - as it was before).

Reply:
The previous public method was incorrect - it is used in the evaluators and would lead to evaluating the non-processed predictions instead of the actual model. The current version of the code fixes this bug.

I would argue that it is expected to get the non-processed results. In fact, bevor this PR the dosctring of the method (falsely) claimed: "...to a result as return by predict". I will adjust the docstring to the original version in my next commit, forgot to do it before.

Here is the previously faulty peace of code that is fixed now:

![image](https://user-images.githubusercontent.com/35432522/105424658-066adf00-5c48-11eb-9f2e-2f7045658556.png)

Reply:
The error is not in `convertClassProbabilitiesToPredictions` but in this evaluator, which does not take into account that there might be an output transformer.
The phrase "a result as returned by predict" does not mean that it is *the* same result but *a* result of the same form.

Reply:
But the issue you found again makes me question whether we should be keeping output transformers.
And target transformers probably make sense only for regressors.

Reply:
I am not a big fan of the output transformers myself, as you know. A quick, non-invasive fix for the issue in the above evaluator would be to move all post processing from the base class to VectorRegressionModel.

If we don't do that, fixing the evaluator would be a bit hacky. _applyPostProcessors is private (and I don't want to make it public) so we would need to recreate it's functionality in the evaluator. This code duplication seems too fragile to me

Reply:
@opcode81 I went ahead and moved the post-processing to regression models only. The code seems clearer to me now. This change is in a single commit so it can easily be reverted, in case you disagree